test/api/models/trial.js

```diff
@@ -300,6 +301,13 @@ describe('Trial', () => {
           .then((trialId) => new Trial({ id: trialId }).fetch({ withRelated: ['documents', 'documents.source'] }))
           .then((trial) => should(trial.toJSON().sources).deepEqual({}))
       ));
+
+      it('returns its own source', () => (
+        factory.create('source', { id: 'nct' })
+          .then(() => factory.create('trialWithRelated', { source_id: 'nct' }))
+          .then((trial) => (should(trial.toJSON().sources).have.key(trial.attributes.source_id))
+          )
+      ));
```

Comment:
On `trialWithRelated` vs `trial`, since the `source_id` I'm using is nowhere in the factory, there's no real danger. But as far as testing goes, I consider both variants equal. I will go back to simple `trial` though, there's no need for relations (anymore, it was at some point in my adventures).

On unloaded `trial.source`, I agree with `undefined` value. Makes sense.

I didn't consider testing an eventual `source` fail, thanks for catching that. It will not escape this time :)